src/chainstate/burn/db/sortdb.rs

```diff
@@ -2494,23 +2550,29 @@ impl SortitionDB {
 
     fn check_schema_version_and_update(&mut self, epochs: &[StacksEpoch]) -> Result<(), db_error> {
         let tx = self.tx_begin()?;
-        match SortitionDB::get_schema_version(&tx) {
-            Ok(Some(version)) => {
-                let expected_version = SORTITION_DB_VERSION.to_string();
-                if version == expected_version {
-                    return Ok(());
-                }
-                if version == "1" {
-                    SortitionDB::apply_schema_2(&tx, epochs)?;
-                    tx.commit()?;
-                    Ok(())
-                } else {
-                    panic!("The schema version of the sortition DB is invalid.")
+        let mut updated = false;
```

Comment:
To future-proof this a bit better, instead of using an `updated` flag, I think it would be safer to do the following:

```rust
let mut cur_version = "1".to_string();
while cur_version != SORTITION_DB_VERSION {
   match SortitionDB::get_schema_version(&tx) {
      Ok(Some(version)) => {
         cur_version = version;
         
         // match on cur_version to decide what schema to apply
         if cur_version == "1" { /* etc. */ }
         else if _cur_version == "2" { /* etc. */ }
      }
      /* other cases */
   }
}
```